packit/sync.py

```diff
@@ -0,0 +1,47 @@
+import glob
+import os
+import shutil
+import logging
+
+from packit.exceptions import PackitException
+from packit.config import PackageConfig, SyncFilesItem, SyncFilesConfig
+
+logger = logging.getLogger(__name__)
+
+
+def get_wildcard_resolved_sync_files(pc: PackageConfig) -> None:
+    logger.debug("Packit synced files %s", pc.synced_files.files_to_sync)
+    files_to_sync = []
+    for sync in pc.synced_files.files_to_sync:
+        if isinstance(sync, str):
+            files_to_sync.append(SyncFilesItem(src=sync, dest=sync))
+            continue
+        src = glob.glob(sync.src)
+        if src:
+            for s in src:
+                files_to_sync.append(SyncFilesItem(src=s, dest=sync.dest))
+    logger.debug(f"Resolved synced file {files_to_sync}")
+    pc.synced_files = SyncFilesConfig(files_to_sync=files_to_sync)
+
+
+def sync_files(pc: PackageConfig, src_working_dir: str, dest_working_dir: str) -> None:
+    """
+    Sync required files from upstream to downstream.
+    """
+    get_wildcard_resolved_sync_files(pc)
+    logger.debug(f"Copy synced files {pc.synced_files.files_to_sync}")
+    for fi in pc.synced_files.files_to_sync:
+        # Check if destination dir exists
+        # If not create the destination dir
+        dest_dir = os.path.join(dest_working_dir, fi.dest)
+        logger.debug(f"Destination {dest_dir}")
+        # Sync all source file
+        src_file = os.path.join(src_working_dir, fi.src)
+        logger.debug(f"Source file {src_file}")
+        if os.path.exists(src_file):
+            logger.info(f"Syncing {src_file}")
+            shutil.copy2(src_file, dest_dir)
```

Comment:
does that mean, that there are not copyied subdirectories? or this expansion is done somewhere above in code?
eg. I have 
```
 - src: fedora/tests
 - dest: tests
```
and in `directory` there are another files and subdirectories, eg `integration`, `unit` they will not be copied?
I've just expected that there will be `shutils.copytree` in these case.

Reply:
Well, python mechanism is used glob. But your case means a file `fedora/tests`.
The code could be improved so that if `src` ends with `/` then append `*` and make glob.

Reply:
no, my question was not about globbing of that, but that shutils.copy2(directory, dest)
is unable to copy directories at all, it copy just regular files

Reply:
Function `get_wildcard_resolved_sync_files` generates only regular files, therefore shutils.copy2 can be used. Even if you specify in `packit.yaml` `src: fedora/tests/`.

Reply:
so that mean, that there is no support for copying directory content recursively. what means that it is no able to sync `tests` for fedora CI
like: 
```
 - src: tests
   dest:tests/upstream_tests
 - src: fedora/tests.yaml
    dest: tests/tests.yaml
```
and then run these upstream tests inside fedora CI.
what I though that it was also purpose of this PR.

It could be workarouded by explicitly mention all subdirecoties, but it will be worse to maintain.

Reply:
after discussion created new issue: https://github.com/packit-service/packit/issues/233 for this item